### PR TITLE
Limit max version of dry-validation 

### LINKED ISSUE #222: dry-validation 1.x moves schema support to dry-schema
Updating to dry-validation 1.x (which the gem spec allows) breaks config when `config.schema` is used with:

```
undefined method `Schema' for Dry::Validation:Module (NoMethodError)
```

It appears that the schema support has been moved to a new `dry-schema` gem so I guess some tweaking is needed to support that...

### PR DESCRIPTION
**Pull request by zbelzer**
Version 1.0 is a breaking change:
https://github.com/dry-rb/dry-validation/blob/master/CHANGELOG.md#v100-2019-06-10

Fixes #222

### CODE REVIEW

**Comment by zbelzer** [comment]
OK, I'm pretty sure I can't make code climate happy since it thinks a line is too long, but if I put it on multiple it tells me it wants it on one again.

In lieu of any rewrites, can we please get a quick win on limiting the version?

File: spec/app/rails_5/config/initializers/new_framework_defaults.rb [hunk 1]
```diff
@@ -17,8 +17,9 @@
 # Require `belongs_to` associations by default. Previous versions had false.
 Rails.application.config.active_record.belongs_to_required_by_default = true
 
+# Removed in rails 5.2+
```
Comment (pkuczynski) [inline spec/app/rails_5/config/initializers/new_framework_defaults.rb]: This was needed only for tests for Rails 5.0. There is a separate 5.2 folder. So this change should not be required?

File: lib/config/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Config
-  VERSION = '1.7.1'
+  VERSION = '1.7.2'.freeze
```
Comment (pkuczynski) [inline lib/config/version.rb]: You should not bump the version. `freeze` is OK

File: spec/app/rails_5/config/initializers/new_framework_defaults.rb (hunk 1 above)
Comment (zbelzer) [reply spec/app/rails_5/config/initializers/new_framework_defaults.rb]: It blows up on `rake spec`. That must run the wrong combination of directories and versions? I had trouble getting appraisal to work here.

Comment (pkuczynski) [reply spec/app/rails_5/config/initializers/new_framework_defaults.rb]: Hmm, made some changes. Lets see if build fails.

File: lib/config/version.rb (hunk 2 above)
Comment (zbelzer) [reply lib/config/version.rb]: I figured it needed to be bumped so bundler recognizes a new version when it's pushed. I'm used to doing it myself for work, forgot that might be part of a different process elsewhere.

File: spec/app/rails_5/config/initializers/new_framework_defaults.rb (hunk 1 above)
Comment (zbelzer) [reply spec/app/rails_5/config/initializers/new_framework_defaults.rb]: Totally happy if it's not needed.

File: config.gemspec [hunk 3]
```diff
@@ -25,11 +25,9 @@ Gem::Specification.new do |s|
 
   s.add_dependency 'activesupport',  '>= 3.0'
   s.add_dependency 'deep_merge',     '~> 1.2',   '>= 1.2.1'
+  
   s.add_dependency 'dry-validation', '~> 0.12', '>= 0.12.2', '< 1.0.0' if RUBY_VERSION >= '2.2'
-
-  if RUBY_VERSION >= '2.1' && RUBY_VERSION < '2.2'
-    s.add_dependency 'dry-validation', '~> 0.10', '>= 0.10.7', '< 1.0.0'
-  end
+  s.add_dependency 'dry-validation', '~> 0.10', '>= 0.10.7', '< 1.0.0' if RUBY_VERSION >= '2.1' && RUBY_VERSION < '2.2'
```
Comment (zbelzer) [inline config.gemspec]: Code climate complained that this line was too long. This is how I had it originally.

File: lib/config/version.rb (hunk 2 above)
Comment (pkuczynski) [reply lib/config/version.rb]: I will upgrade this whenever the gem will get released. Saying that, I can release once this PR is merged I guess.

File: config.gemspec (hunk 3 above)
Comment (pkuczynski) [reply config.gemspec]: Does not seem to complain anymore?

**Comment by pkuczynski** [comment]
Hmm it now fails in many places due sqlite issue. Would you have some time to debug?

**Comment by zbelzer** [comment]
Sure, but it will probably be in the morning.

File: Appraisals [hunk 4]
```diff
@@ -1,34 +1,42 @@
 appraise 'rails-3' do
+  gem 'sqlite3', "< 1.4.0"
```
Comment (pkuczynski) [inline Appraisals]: Could you use single quotes instead of double please?

**Comment by pkuczynski** [comment]
We are getting very close! Still few things failing I am afraid :(

File: Appraisals (hunk 4 above)
Comment (zbelzer) [reply Appraisals]: Whoops

**Comment by zbelzer** [comment]
@pkuczynski Where are there always new code climate failures?

**Comment by zbelzer** [comment]
OMGeee. I got the tests to pass. @pkuczynski Can we tell code climate to ignore those string macros or do you need me to add them.

**Comment by pkuczynski** [comment]
Code climate fixed :)

**Comment by zbelzer** [comment]
@pkuczynski all good?

**Comment by pkuczynski** [comment]
Looks good to me! :) Should I release it? You also put a lot of effort into this PR - would you like to become a contributor?

**Comment by zbelzer** [comment]
Please release it!

**Comment by pkuczynski** [comment]
Already released :)

**Comment by pkuczynski** [comment]
How about the contribution part?

**Comment by zbelzer** [comment]
Sure. That would be great.

**Comment by pkuczynski** [comment]
Done! :) Welcome onboard :)

**Pull request merged**